activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -115,7 +115,7 @@ def _decrypt(encrypted_message)
         # Currently the OpenSSL bindings do not raise an error if auth_tag is
         # truncated, which would allow an attacker to easily forge it. See
         # https://github.com/ruby/openssl/issues/63
-        raise InvalidMessage if aead_mode? && auth_tag.bytes.length != 16
+        raise InvalidMessage if aead_mode? && (auth_tag.nil? || auth_tag.bytes.length != 16)
```

Comment:
Why do we need the `nil?` check? Should that be a separate fix + regression tests we backport to 5.1 and 5.0 as well?